crates/ruff_python_formatter/tests/fixtures.rs

```diff
@@ -492,3 +531,49 @@ source_type                = {source_type:?}"#,
         )
     }
 }
+
+/// Collects the unsupported syntax errors and assigns a unique hash to each error.
+fn collect_unsupported_syntax_errors(
+    errors: &[UnsupportedSyntaxError],
+) -> FxHashMap<u64, UnsupportedSyntaxError> {
+    let mut collected = FxHashMap::default();
+
+    for error in errors {
+        let mut error_fingerprint = fingerprint_unsupported_syntax_error(error, 0);
+
+        // Make sure that we do not get a fingerprint that is already in use
+        // by adding in the previously generated one.
+        loop {
+            match collected.entry(error_fingerprint) {
+                Entry::Occupied(_) => {
+                    error_fingerprint =
+                        fingerprint_unsupported_syntax_error(error, error_fingerprint);
+                }
+                Entry::Vacant(entry) => {
+                    entry.insert(error.clone());
+                    break;
+                }
+            }
```

Comment:
This is inspired by our gitlab reporter that uses the very same approach for lint errors.